packages/nodes-base/nodes/Google/Sheet/v2/actions/sheet/update.operation.ts

```diff
@@ -380,6 +381,10 @@ export async function execute(
 		if (!updateData.length) {
 			return [];
 		}
-		return this.helpers.returnJsonArray(mappedValues);
+		const itemData = generatePairedItemData(items.length);
+		const executionData = this.helpers.constructExecutionMetaData(wrapData(mappedValues), {
+			itemData,
+		});
+		return executionData;
```

Comment:
These should map the matching incoming item instead of setting all input items as pairedItem.